pypeit/core/tracewave.py

```diff
@@ -693,41 +701,52 @@ def fit_tilts(trc_tilt_dict, thismask, slit_cen, spat_order=3, spec_order=4, max
     # Embed the old thismask in the new larger padded thismask
     thismask_pad[ind_spec + pad_spec, ind_spat + pad_spat] = thismask[ind_spec, ind_spat]
     # Now grow the thismask_pad
-    kernel = np.ones((2*pad_spec, 2*pad_spat)) / float(4 * pad_spec * pad_spat)
+    kernel = np.ones((2 * pad_spec, 2 * pad_spat)) / float(4 * pad_spec * pad_spat)
     thismask_grow = ndimage.convolve(thismask_pad.astype(float), kernel, mode='nearest') > 0.0
     # Evaluate the tilts on the padded image grid
     tiltpix = spec_img_pad[thismask_grow] + xnspecmin1 \
-                    * utils.func_val(coeff2, spec_img_nrm[thismask_grow], func2d,
-                                     x2=dspat_img_nrm[thismask_grow], minx=0.0, maxx=1.0,
-                                     minx2=-1.0, maxx2=1.0)
+              * utils.func_val(coeff2, spec_img_nrm[thismask_grow], func2d,
+                               x2=dspat_img_nrm[thismask_grow], minx=0.0, maxx=1.0,
+                               minx2=-1.0, maxx2=1.0)
 
     # Now do one last fit to invert the function above to obtain the
     # final tilts model in normalized image coordinates
     inmask = np.isfinite(tiltpix)
     sigma = np.full_like(spec_img_pad, 10.0)
+
+    # Avoid substantial extrapolation..
+    low = spec_img_pad[thismask_grow] < np.min(tilts_spec - min_extrap)
+    tiltpix[low] = spec_img_pad[thismask_grow][low]
+    sigma[thismask_grow][low] = sigma[thismask_grow][low] * 10.
+    # TODO -- Condsider adding in this anti-extrapolation code too
+    #high = spec_img_pad[thismask_grow] > np.max(tilts_spec + 20.)
+    #tiltpix[high] = spec_img_pad[thismask_grow][low]
+    #sigma[thismask_grow][high] = sigma[thismask_grow][high] * 10.
```

Comment:
Done.  now named `minmax_extrap`

Reply:
I've set the high value to 1000. for now on the grounds of 
it ain't broke, don't fix it.

Reply:
Not to belabor the issue, but you could have the code accept `None` elements of `minmax_extrap`, and use that to signify that no limit should be set.

Reply:
As for an instrument, this was introduced because of the Dev suite failure of LRISr R400 
with numpy > 16.0.  I have run on the full `develop` Dev suite and see no (obvious) harmful effects in the outputs.